solidity/contracts/bridge/Bridge.sol

```diff
@@ -771,6 +742,53 @@ contract Bridge is Ownable, EcdsaWalletOwner {
         // TODO: Handle deposits having `vault` set.
     }
 
+    /// @notice Resolves sweeping wallet based on the provided wallet public key
+    ///         hash. Validates the wallet state and current main UTXO, as
+    ///         currently known on the Ethereum chain.
+    /// @param walletPubKeyHash public key hash of the wallet proving the sweep
+    ///        Bitcoin transaction.
+    /// @param mainUtxo Data of the wallet's main UTXO, as currently known on
+    ///        the Ethereum chain. If no main UTXO exists for the given wallet,
+    ///        this parameter is ignored
+    function resolveWallet(
```

Comment:
That function should be used only during the sweep. What do you think about naming it `resolveSweepingWallet`?

Reply:
Yep. 60114020a0dda22a3210901e720af574312cb946